pzsd_bot/__main__.py

```diff
@@ -1,48 +1,214 @@
+                    select(pzsd_user).where(
+                        pzsd_user.c.discord_snowflake == recipient_id
+                    )
+                )
+            try:
+                recipient = result.one_or_none()
+            except MultipleResultsFound:
+                recipient_type = "name" if recipient_name else "snowflake"
+                logger.error(
+                    "Multiple users with %s '%s' found in user table",
+                    recipient_type,
+                    recipient_name or recipient_id,
+                )
+                raise
+
+            if recipient is None:
+                logger.info(
+                    "%s tried to bestow points to '%s' but they weren't in the user table",
+                    bestower.name,
+                    recipient_name or recipient_id,
+                )
+                return
+
+        excessive_point_violation = abs(point_amount) > POINT_MAX_VALUE
+        if excessive_point_violation:
+            logger.info(
+                "%s tried to give %s more than the max allowed points (%s)",
+                bestower.name,
+                recipient.name,
+                format(point_amount, ','),
+            )
+            return
+
+        self_point_violation = bestower.id == recipient.id
+        if self_point_violation:
+            logger.info(
+                "%s attempted to give themselves %s points. Very naughty.",
+                bestower.name,
+                point_amount,
+            )
+        else:
+            logger.info(
+                "%s awarding %s point(s) to %s",
+                bestower.name,
+                format(point_amount, ','),
+                recipient.name,
+            )
+
+        if not self_point_violation:
+            async with engine.begin() as conn:
+                await conn.execute(
+                    insert(ledger).values(
+                        bestower=bestower.id,
+                        recipient=recipient.id,
+                        points=point_amount,
+                    )
+                )
+                logger.info("Added point transaction to ledger")
+
+        if self_point_violation:
+            title = "Self point violation!"
+            color = 0xFF0000
+        else:
+            title = "Point transaction"
+            color = 0xFFFFFF
+        embed = discord.Embed(
+            title=title,
+            description=f"[Jump to original message]({message.jump_url})",
+            colour=color,
+            timestamp=datetime.now(),
+        )
+        embed.add_field(name="Bestower", value=bestower.name, inline=True)
+        embed.add_field(name="Recipient", value=recipient.name, inline=True)
+        embed.add_field(
+            name="Point amount", value=format(point_amount, ","), inline=True
+        )
+        message_content = message.content
+        if len(message_content) > 80:
+            message_content = message_content[:80] + "..."
```

Comment:
Use one U+2026 unicode ellipses character `…` rather than three U+002E full stops

Reply:
lmao why not